src/Cli/dotnet/NugetPackageInstaller/NuGetPackagePathResolver.cs

```diff
@@ -0,0 +1,25 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using NuGet.Packaging;
+using NuGet.Packaging.Core;
+
+namespace Microsoft.DotNet.Cli.NuGetPackageInstaller
+{
+    internal class NuGetPackagePathResolver : PackagePathResolver
+    {
+        public NuGetPackagePathResolver(string rootDirectory) : base(rootDirectory, false)
+        {
+        }
+
+        public override string GetPackageDirectoryName(PackageIdentity packageIdentity)
+        {
+            return string.Empty; // Extract package directly to target directory
+        }
+
+        public override string GetPackageFileName(PackageIdentity packageIdentity)
```

Comment:
Why this need an override. The base class is https://github.com/NuGet/NuGet.Client/blob/f24bad0668193ce21a1db8cabd1ce95ba509c7f0/src/NuGet.Core/NuGet.Packaging/PackagePathResolver.cs#L57
```
public virtual string GetPackageFileName(PackageIdentity packageIdentity)
        {
            var fileNameBase = GetPathBase(packageIdentity);

            fileNameBase.Append(PackagingCoreConstants.NupkgExtension);

            return fileNameBase.ToString();
        }
```

They do not do the same?

Reply:
The base class adds some extra directories to the target path. I overrode it so we could specify the directory where the nupkg would go directly.

Reply:
Yes, exactly, I'll add a comment.